src/sync/incomingSyncUtils.js

```diff
@@ -674,11 +674,15 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
     case 'Store': {
       const { tags, custom_data } = record;
       if (settings.get(THIS_STORE_ID) === record.ID) {
-        database.update('Setting', { key: THIS_STORE_TAGS, value: tags });
-        database.update('Setting', {
-          key: THIS_STORE_CUSTOM_DATA,
-          value: validateJson(custom_data),
-        });
+        if (validateJson(custom_data)) {
+          database.update('Setting', {
+            key: THIS_STORE_CUSTOM_DATA,
+            value: custom_data,
+          });
+        }
+        if (validateJson(tags)) {
```

Comment:
tags are not json btw, just a text string. could also be null though

Also your idea in the PR description sounds like the way to go validateJson -> to return empty string.

Reply:
b23db3d9df301ae8d3341571b58767f6b17fcf68

The real problem is 4D sending a string 'undefined' which throws and error during parsing. 

My reluctance was setting fields as an empty string rather than being ignored. As well as `''` being falsey

Reply:
sure, I was just thinking of a use case where customData is fully removed, was wondering if 4D would send null or 'undefined', then nothing will change in settings sync if it's just ignored